sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -216,6 +217,28 @@ async def __anext__(self):
             chunk = await self.response.internal_response.content.read(self.block_size)
             if not chunk:
                 raise _ResponseStopIteration()
+            if not self._decompress:
+                return chunk
+            enc = self.response.internal_response.headers.get('Content-Encoding')
+            if not enc:
+                return chunk
+            enc = enc.lower()
+            if enc in ("gzip", "deflate", "br"):
```

Comment:
I think I'd prefer for a user to be able to register content encodings they can handle. And us not advertising support for brotli encodings. I am not entirely comfortable taking  a "hidden" dependency on `brotlipy`.

Reply:
This is the behavior in aiohttp. I think we don't want to change the behavior, right?

Reply:
Thinking. It's not that I want to change the behavior (in fact, I very much don't want to do that). But the combination we are talking about here is range requests + auto decoding + brotli. Which is rare. And the user does have the work-around for that (raw stream + decompress themselves). I want to do a bit more due diligence before *we* directly recommend brotlipy to our users (there is a subtle difference)

Reply:
I think the scenario is non-range requests + auto decoding + 'br' content-encoding.

Currently, it succeeds and user gets decoded stream.

If we remove 'br' here, user will get encoded stream which is a behavior change, right?

Reply:
And brotlipy is what aiohttp uses. aiohttp also "hiddenly" import brotlipy. So I think it is fair to have it in our aiohttp transport.

Reply:
And this is aiohttp's code:

https://github.com/aio-libs/aiohttp/blob/master/aiohttp/http_parser.py#L44

https://github.com/aio-libs/aiohttp/blob/master/aiohttp/http_parser.py#L846

Reply:
Ok, a couple of things to point out -  aiohttp depends on [`Brotli`](https://pypi.org/project/Brotli/), not [`brotlipy`](https://pypi.org/project/brotlipy/). Which makes a pretty big difference in my mind. Secondly, aiohttp has made a decision to include a set of things as extras. Which we really should do as well if we want to advertise support for plug-ins like is the case here. 

If it wasn't for the potential of breaking existing working workloads, I would have strongly suggested a different pattern here. As it stands, I'm not sure what to do that isn't breaking unless we mimic what aiohttp does.

Reply:
...but, we should absolutely make sure that we suggest that users install the right dependency in the right way!

Reply:
...and we should add tests for brotli encoded responses.

Reply:
Took offline.